### DISCUSSION TITLE
Allow inherited authentication that is configured at collection or directory level

### DISCUSSION THREAD

**Discussion by andreassiegel** [Kong/insomnia#6205] [Ideas] [answered]
When working with different APIs, many (or even all) requests share the same authentication. Right now, it has to be configured for each request individually although all actually require the exact same settings.
These settings, however, are not the same for all requests in all directories in a collection.

For instance, there are APIs that require HTTP Basic Authentication, others use API keys or OAuth 2.0 with client credentials or authorization code flow. Actual credentials might also differ between requests or APIs. So there is quite some variance but still, requests have a lot in common, and some grouping mechanism would be appreciated.

This actually also is an issue with the collection import from Postman as Postman allowed the inheritance of authentication from directories but that information is gone so requests will not work and have to be updated manually.

Therefore, it would be good to be able to set the authentication only once and be able to configure individual requests to either use their own authentication configuration or inherit the configuration from a parent.

This feature request certainly is related to https://github.com/Kong/insomnia/discussions/3694 but in my mind not the same as the other is more getting at environments, and I would like to see that as part of the collection configuration. Also, that discussion is quite old already.

**Comment by pczopa** [comment]
Migrating from Postman to Insomnia and this is the most annoying thing to have to deal with - would love for this feature to be included in Insomnia!

**Comment by subnetmarco** [comment]
I am prioritizing this for the April/May timeframe, after the large Insomnia 9.x release we are about to release in March.

**Comment by subnetmarco** [comment, marked as answer]
This is possible since [Insomnia 9.3](https://konghq.com/blog/product-releases/insomnia-9-3-ga).

**Comment by K3UL** [comment]
@subnetmarco this is **partially** possible. It is now possible to do it at the directory level (which is already a huge improvement, thank you for that).

But it is not possible to do it at the collection level, which means that if I have a collection where the auth is the same, I still have to create a parent directory, just to carry certain settings.

Only a minor inconvenience, but it would be nice to also have the settings on the collection.

**Reply by subnetmarco** [reply]
> I still have to create a parent directory, just to carry certain settings.

Yes, I agree that while this provides a solution, it still requires that parent folder. From a UX standpoint, where do you think we should add these "Collection Level" settings? It's actually not intuitive where it should be.

If it's hidden in this dropdown (where it should naturally be placed), nobody will be able to see it:

![Screenshot 2024-10-03 at 11 28 16 AM](https://github.com/user-attachments/assets/d16b2a42-2c48-47ac-affa-cf583043c042)

Or maybe it can be a default (not removable) element at the top of the list here:

![Screenshot 2024-10-03 at 11 34 52 AM](https://github.com/user-attachments/assets/c8c650ee-7337-4591-854f-e2ead0522b7a)

What do you think? @K3UL

**Reply by mathisgauthey** [reply]
> > I still have to create a parent directory, just to carry certain settings.
> 
> Yes, I agree that while this provides a solution, it still requires that parent folder. From a UX standpoint, where do you think we should add these "Collection Level" settings? It's actually not intuitive where it should be.
> 
> If it's hidden in this dropdown (where it should naturally be placed), nobody will be able to see it:
> 
> ![Screenshot 2024-10-03 at 11 28 16 AM](https://private-user-images.githubusercontent.com/964813/373104159-d16b2a42-2c48-47ac-affa-cf583043c042.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNDgwNDgsIm5iZiI6MTczNjM0Nzc0OCwicGF0aCI6Ii85NjQ4MTMvMzczMTA0MTU5LWQxNmIyYTQyLTJjNDgtNDdhYy1hZmZhLWNmNTgzMDQzYzA0Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwOFQxNDQ5MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMjRkNGRmNWE2YzhjZjhkYjc1M2U4MTc0NTVlZGQ2NzA4MjM4NzVlMmRiNWI3YTRmNDQwOTIzYThjYWVhYWU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qijloiNBGvdP3sEx9rinRLdnASRRfA-B4h9WT3nsq_A)
> 
> Or maybe it can be a default (not removable) element at the top of the list here:
> 
> ![Screenshot 2024-10-03 at 11 34 52 AM](https://private-user-images.githubusercontent.com/964813/373105286-c8c650ee-7337-4591-854f-e2ead0522b7a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNDgwNDgsIm5iZiI6MTczNjM0Nzc0OCwicGF0aCI6Ii85NjQ4MTMvMzczMTA1Mjg2LWM4YzY1MGVlLTczMzctNDU5MS04NTRmLWUyZWFkMDUyMmI3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwOFQxNDQ5MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xN2RlY2YwOTc2YjY5OGQ4M2IxYzFiNjhjZGNiMWE4MGM3MDAyY2Q3OGRjOGY1NDlmNzVhODMzOTM0YmQ0OTA3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rj7tujAVeCyLLXp8dF5npywaqhzVlmqo4voZ3YHyCCk)
> 
> What do you think? @K3UL

Hey there, is there any update on that feature ? Can we use collection level settings now or is it still on folders level ?

Thanks in advance !

**Comment by K3UL** [comment]
True this is a challenge unique to the way it is organized in Insomnia, I appreciate you taking the time to consider it.

Well I am no UX specialist but yes I think your second solution is better. Another option IMO would be to have this settings button next to the Runner button 
![CleanShot 2024-10-03 at 11 23 46@2x](https://github.com/user-attachments/assets/961869ae-96de-4a4a-9a1d-5ce039f79c58)

And it could display the settings on the right just like the directory settings.

Or also, the item "Collection settings" like you added but instead of having it at the top of the requests, it could be with "Manage cookies" and "Manage certificates"

**Comment by vbalashi** [comment]
Does this work? I can neither add Auth to collection, nor to Folder. Neither can find anything in the documentation.

**Reply by ryan-willis** [reply]
You can configure folder auth by clicking on a folder:

![folder-auth](https://github.com/user-attachments/assets/d2f1a925-7721-4bec-80b2-83808ee6cdfb)

**Comment by av1v3k** [comment]
hmm..... still not resolved ? Instead of "Inherit from Parent" ,I use currently from base custom environment, which I had defined a variable in it.

**Reply by mathisgauthey** [reply]
I'd suggest taking a look at the open source alternative that is Bruno 👌
It has a proper collection and folder level management of variables, auths, tests, etc.

I mean no offense by pointing to it here but as the situation has not yet evolved and the two of them are open source, citing them might help either way. Either directing users to another tool with that feature, or increasing the projected value of that feature in Insomnia.
